### PR TITLE
add 24h total volume

### PR DESCRIPTION
**Pull request by devexpert127**

### CODE REVIEW

File: src/components/OptionMarket/index.tsx [hunk 1]
```diff
@@ -135,6 +134,65 @@ export const OptionMarket = () => {
     });
   }
 
+  async function get24HRTotalVolume(assetPoolList: poolByMint) {
+    const keys = Object.keys(assetPoolList);
+    const poolList = keys.map(key => '"' + key + '"');
+    
+    const query = gql`
+      {
+        tokensStats(addresses: [
```
Comment (tomjohn1028) [inline src/components/OptionMarket/index.tsx]: This is getting volume for the spot markets. We want the volume for the active PsyOptions markets. 

To get the `ARRAY_OF_SERUM_MARKET_ADDRESSES` you can do the following:

1. Given all OptionMarkets (as loaded from the TVL task), derive their Seurm market addresses
    * See example from PsyOptions UI [here](https://github.com/mithraiclabs/psyoptions-frontend/blob/staging/src/hooks/useDeriveMultipleSerumMarketAddresses.ts#L17-L23)
2. Group the Serum address as follows
_ETH-CALLS_: any `OptionMarket` where the _underlying_asset_mint_ is the ETH mint
_ETH-PUTS_: any `OptionMarket` where the _quote_asset_mint_ is the ETH mint
_BTC-CALLS_: any `OptionMarket` where the _underlying_asset_mint_ is the BTC mint
_BTC-PUTS_: any `OptionMarket` where the _quote_asset_mint_ is the BTC mint
...and so on grouping other unknown mints that show up in a similar fashion, ignoring USDC
3. For each group above query for the Serum stats with the query below (be mindful of rate limits as this list will grow)

```
{
  dailyStats( markets: ARRAY_OF_SERUM_MARKET_ADDRESSES )
    {
    stats {
      au1h
      vol1hUsd
      au24h
      vol24hUsd
      au7d
      vol7dUsd
    }
  }
}
```

**Review by tomjohn1028 — CHANGES_REQUESTED** [review]

**Comment by devexpert127** [comment]
Error: Component must be wrapped with <Container.Provider>
when using useDeriveMultipleSerumMarketAddresses

**Comment by tomjohn1028** [comment]
> Error: Component must be wrapped with <Container.Provider>
when using useDeriveMultipleSerumMarketAddresses

You are in a separate codebase so the hook cannot be used out of the box. It was more for an example, but that code is in flux with some protocol updates made this weekend XD apologies haha. Here's updated line number https://github.com/mithraiclabs/psyoptions-frontend/blob/staging/src/hooks/useDeriveMultipleSerumMarketAddresses.ts#L23-L41

**Comment by devexpert127** [comment]
quoteMint has null in default publickey
const _quoteMint = useRecoilValue(quoteMint);
_quoteMint is null in useDeriveMultipleSerumMarketAddresses
https://github.com/mithraiclabs/bi-dashboard/blob/c3c423802b9e32f4c831760dabf4ca1a8b46c51a/src/hooks/useDeriveMultipleSerumMarketAddresses%20.ts#L15

**Comment by devexpert127** [comment]
image
This is update version of 24hr and 7d total volumes

**Review by tomjohn1028 — APPROVED** [review]
@devexpert127 Looks great! I only saw the TVL and 7 day volume show up on the UI though, might be missing the 24hr

**Comment by devexpert127** [comment]
It has 24hr too, I uploaded only 24hr and 7d now.

**Comment by devexpert127** [comment]
image

**Comment by devexpert127** [comment]
https://github.com/mithraiclabs/bi-dashboard/pull/8

I've updated group by calls and puts

**Pull request merged**